packages/core/src/js/feedback/FeedbackForm.tsx

```diff
@@ -53,11 +54,15 @@ export class FeedbackForm extends React.Component<FeedbackFormProps, FeedbackFor
       associatedEventId: eventId,
     };
 
-    onFormClose();
-    this.setState({ isVisible: false });
-
-    captureFeedback(userFeedback);
-    Alert.alert(text.successMessageText);
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    checkInternetConnection(() => {
+      onFormClose();
+      this.setState({ isVisible: false });
+      captureFeedback(userFeedback);
+      Alert.alert(text.successMessageText);
+    }, () => {
+      Alert.alert(text.errorTitle, text.networkError);
+    });
```

Comment:
I think the only negative side with this implementation is that the user will not have an opportunity to retry sending the feedback in case of internet failure.

In this case, we'd need to invert our logic again by closing the form only after captureFeedback succeeds.

Reply:
Good point 👍  

The current implementation should close the form only on success. In case there is a network issue the error alert pops on top of the form and the user can close it and retry sending the feeback. Do you mean adding a 'Retry' button?

Reply:
after reading the other part of the code, this will not be an issue since we will only try to send the feedback and closing it only after making sure we can communicate with the server.